src/components/PopupWithForm.js

```diff
@@ -5,6 +5,19 @@ class PopupWithForm extends Popup {
     super({ popupSelector });
     this._handleFormSubmit = handleFormSubmit;
     this._popupForm = this._popupElement.querySelector(".modal__form");
+    console.log(this._popupElement); // Check if this logs the correct element
+    if (!this._popupElement) {
+      throw new Error(`Element not found for selector: ${selector}`);
+    }
+  }
+
+  setLoading(isLoading) {
+    this._submitBtn = this._popupElement.querySelector(".modal__button");
+    if (isLoading) {
+      this._submitBtn.textContent = "Saving...";
+    } else {
+      this._submitBtn.textContent = "Submit";
```

Comment:
`"Submit"` is not the text of the buttons according to the design. It should be `Save`


**COULD BE IMPROVED**

It’s better to make a `renderLoading` method inside `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```